arduino/WandCore/WandCore.ino

```diff
@@ -253,72 +257,114 @@ void loop()
   int currentTouchEvent = getTouchEvent();
-  switch (currentTouchEvent)
-  {
-    case SINGLE_TAP:
-      // trigger color sensor
-      debugI("single tap happened");
-      
-      break;
-    case DOUBLE_TAP:
-      debugI("double tap happened");
-      // change mode
-      if (wandMode == POWER) {
-        wandMode = COLOR;
-        incrementRGBColorloop(); 
-        debugI("Switched to Color Mode");
-      } else {
-        wandMode = POWER;
-        setRGBLED(255, 255, 255); // white for power mode
-        debugI("Switched to Power Mode"); 
-      }      
-      
-      break;
-    case HOLDING:
-      // TODO adjust brightness or color depending on mode
-      if ((millis() - lastLogged) >= 500) { // DEBUG ONLY, throttling log output
-        debugI("touch is being held");
-        lastLogged = millis();
-      }
+  if (currentTouchEvent == SINGLE_TAP) {
+    debugI("Single Tap Just Happened");
+    if (wandMode == POWER) {                      // power mode, single tap == set light brightness based on "brightness" of color sensed
+      // TODO
+    }
+    else if (wandMode == COLOR) {                 // color mode, single tap == set color of light based on color sensed
+      // TODO
+    }
       
-      break;
+  } else if (currentTouchEvent == DOUBLE_TAP) {
+    debugI("Double Tap Just Happened");
+    // toggle the wand's mode
+    if (wandMode == POWER) {                      // power mode, double tap == switch to color mode
+      wandMode = COLOR;
+      incrementRGBColorloop();                    // resume cycling the onboard RGB through its colorloop
+      debugI("Switched to Color Mode");
+    } else {                                      // color mode, double tap == switch to power mode
+      wandMode = POWER;
+      setRGBLED(255, 255, 255);                   // solid white for power mode
+      debugI("Switched to Power Mode"); 
+    }
   }
-  
-  // Read IMU sensor
-  readIMU(imuData);
 
-  // Calculate sum of acceleration and compare to threshold
+  // Calculate sum of acceleration to compare to threshold
   float aSum = fabs(imuData[AX]) + fabs(imuData[AY]) + fabs(imuData[AZ]);
   float gSum = fabs(imuData[GX]) + fabs(imuData[GY]) + fabs(imuData[GZ]);
 
-  // Acceleration threshold reached, predict gesture
+  // Acceleration threshold reached, handle gesture
   if (aSum >= accelerationThresholdG || gSum >= gyroThreshold) {
+    // do the gesture predicting
     readAndPredictGesture(imuData, gestureConfidence);
     
-    if (gestureConfidence[GESTURE_FLICK] > 0.9) { // it's a flick, do flick things!
+    // read target light state
+    // only do this here as it's not needed elsewhere and this minimizes polling
+    // step 1 of 3 - prepare the raw string from the Hue library
+    String rawLightState;
+    rawLightState = myHue.getLightInfo(lightID);
+    int removeToIdx = rawLightState.indexOf("{") - 1;
+    rawLightState.remove(0, removeToIdx);

```

Comment:
Why is this needed?

Reply:
because the Hue client provides a string of literally the entire HTTP response, including all the headers etc. This cuts the string down to just the JSON string.

I can add comments to capture this in the code.